lib/copy/copy.js

```diff
@@ -122,43 +108,47 @@ async function setDestTimestampsAndMode (srcMode, src, dest) {
   return setDestMode(dest, srcMode)
 }
 
+function fileIsNotWritable (srcMode) {
+  return (srcMode & 0o200) === 0
+}
+
+function makeFileWritable (dest, srcMode) {
+  return setDestMode(dest, srcMode | 0o200)
+}
+
 function setDestMode (dest, srcMode) {
   return fs.chmod(dest, srcMode)
 }
 
 async function onDir (srcStat, destStat, src, dest, opts) {
+  // the dest direcotry might not exist, create it
   if (!destStat) {
     await fs.mkdir(dest)
   }
-  await copyDir(src, dest, opts)
-  if (!destStat) {
-    await setDestMode(dest, srcStat.mode)
-  }
-}
 
-async function copyDir (src, dest, opts) {
   const items = await fs.readdir(src)
-  return copyDirItems(items, src, dest, opts)
-}
 
-function copyDirItems (items, src, dest, opts) {
-  const item = items.pop()
-  if (!item) return
-  return copyDirItem(items, item, src, dest, opts)
-}
+  // loop through the files in the current directory to copy everything
+  for (let i = 0, len = items.length; i < len; i++) {
+    const item = items[i]
```

Comment:
Why not use a `for..of` loop here? Sorry if my initial review suggested I wanted a plain `for` loop.

Reply:
Changed in [`1ce5d34` (#1021)](https://github.com/jprichardson/node-fs-extra/pull/1021/commits/1ce5d348c426ea20c5006389477e163211898b22).